rotkehlchen/tests/unit/test_cost_basis.py

```diff
@@ -790,132 +827,45 @@ def test_accounting_average_cost_basis(accountant):
-        rate=FVal(200),
-        index=4,
-    )
-    asset_events.acquisitions_manager.add_acquisition(event3)
-    assert asset_events.acquisitions_manager.remaining_amount == FVal(1)
-    assert asset_events.acquisitions_manager.current_average_cost_basis == FVal(100)
-    asset_events.acquisitions_manager.add_acquisition(event4)
-    assert asset_events.acquisitions_manager.remaining_amount == FVal(2)
-    assert asset_events.acquisitions_manager.current_average_cost_basis == FVal(300)
-    cost_basis_result = asset_events.acquisitions_manager.calculate_spend_cost_basis(
-        spending_amount=FVal(0.5),
-        spending_asset=asset,
-        timestamp=0,
-        missing_acquisitions=cost_basis.missing_acquisitions,
-        used_acquisitions=asset_events.used_acquisitions,
-        settings=cost_basis.settings,
-        timestamp_to_date=cost_basis.timestamp_to_date,
-        average_cost_basis=asset_events.acquisitions_manager.current_average_cost_basis,
-    )
-    assert asset_events.acquisitions_manager.remaining_amount == FVal(1.5)
-    assert asset_events.acquisitions_manager.current_average_cost_basis == FVal(225)
-    assert cost_basis_result.is_complete is True
-    assert cost_basis_result.taxable_bought_cost == FVal(225)
-    asset_events.acquisitions_manager.add_acquisition(AssetAcquisitionEvent(
-        amount=FVal(0.5),
-        timestamp=5,
-        rate=FVal(500),
-        index=5,
-    ))
-    assert asset_events.acquisitions_manager.remaining_amount == FVal(2)
-    assert asset_events.acquisitions_manager.current_average_cost_basis == FVal(475)
-
-    # see that using more than the available acquisitions adds a MissingAcquisition
-    assert asset_events.acquisitions_manager.calculate_spend_cost_basis(
-        spending_amount=FVal(3.5),
-        spending_asset=asset,
-        timestamp=0,
-        missing_acquisitions=cost_basis.missing_acquisitions,
-        used_acquisitions=asset_events.used_acquisitions,
-        settings=cost_basis.settings,
-        timestamp_to_date=cost_basis.timestamp_to_date,
-        average_cost_basis=asset_events.acquisitions_manager.current_average_cost_basis,
-    ).is_complete is False
-    assert asset_events.acquisitions_manager.remaining_amount == ZERO
-    # it is negative due to the missing acquisition.
-    assert asset_events.acquisitions_manager.current_average_cost_basis == FVal(-356.25)
-    assert cost_basis.missing_acquisitions == [
-        MissingAcquisition(
-            asset=asset,
-            time=0,
-            found_amount=FVal(2),
-            missing_amount=FVal(1.5),
-        ),
-    ]
+    manager = cost_basis.get_events(A_ETH).acquisitions_manager
+
+    # Check 2 simple cases - a spend with positive pnl and a spend with negative pnl
+    add_acquisition(pot, amount=FVal(2), price=FVal(10))  # Buy 2 ETH for $10  total acb: $20
+    current_total_acb = events[0].price * events[0].taxable_amount
+    current_amount = events[0].taxable_amount  # 2
```

Comment:
Read this test and ran the formulas by hand. They seem to mostly check out ... except for when the user has a loss. You should have had 1-2 more scenarios of user having a loss (negative pnl).
Do not discard these scenarios please since I checked them. If anything add to them to have at least one more negative PnL (basically a loss) trade. Also perhaps check that trying to sell more than you got is an error?

Also some general comments on making it easier to read.

Please separate all steps more clearly by at least adding an empty newline between each step and naming them. Step 1, Step2 etc.

Also you don't seem to be consistent in asserting values at each step.

At each step assert on the:
1. current amount
2. total_acb
3. pnl.

In some steps you even write it down in comments but not actually assert that it's that.

Below are my calculations by hand which seem to mostly match apart from the negative -15 at the end:

-----
```
1. Buy 2 ETH for 10.
total_acb: 20
amount: 2

2. Sell 1 ETH for 15.
pnl: 15 - 20 / 2 = 5
total_acb: 20 * 1/2 = 10
amount: 1

3. Buy 1 ETH for 30
amount: 2
total_acb: 10 + 30 = 40

4. Sell 1.5 ETH for 40

pnl: 1.5*40 - (40/2)*1.5 = 30.
total_acb: 40 * (0.5 / 2) = 10.
amount: 0.5


5. Buy 3 ETH for 20
total_acb: 10 + 60 = 70
amount: 3.5

6. Sell 2 ETH for 30

pnl: 2*30 - (70/3.5)*2 = 20
total_acb: 70 * (1.5)/3.5 = 30
amount: 1.5

7. Sell 1.5 ETH for 10

pnl: 1.5*10 - (30/1.5)*1.5 = -15.
total_acb: 30 * 0 / 1.5 = 0.
```